tutorials/Context_Menu/README.md

```diff
@@ -124,4 +122,261 @@ fun main() = singleWindowApplication(title = "Context menu") {
+```kotlin
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.text.JPopupTextMenu
+import androidx.compose.foundation.text.LocalTextContextMenu
+import androidx.compose.foundation.text.selection.SelectionContainer
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.awt.ComposePanel
+import androidx.compose.ui.platform.LocalLocalization
+import java.awt.Color
+import java.awt.Component
+import java.awt.Dimension
+import java.awt.Graphics
+import java.awt.event.KeyEvent
+import java.awt.event.KeyEvent.CTRL_DOWN_MASK
+import java.awt.event.KeyEvent.META_DOWN_MASK
+import javax.swing.Icon
+import javax.swing.JFrame
+import javax.swing.JMenuItem
+import javax.swing.JPopupMenu
+import javax.swing.KeyStroke.getKeyStroke
+import javax.swing.SwingUtilities
+import org.jetbrains.skiko.hostOs
+
+fun main() = SwingUtilities.invokeLater {
+    val panel = ComposePanel()
+    panel.setContent {
+        JPopupTextMenuProvider(panel) {
+            Column {
+                SelectionContainer {
+                    Text("Hello, Compose!")
+                }
+
+                var text by remember { mutableStateOf("") }
+
+                TextField(text, { text = it })
+            }
+        }
+    }
+
+    val window = JFrame()
+    window.contentPane.add(panel)
+    window.size = Dimension(800, 600)
+    window.isVisible = true
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun JPopupTextMenuProvider(owner: Component, content: @Composable () -> Unit) {
+    val localization = LocalLocalization.current
+    CompositionLocalProvider(
+        LocalTextContextMenu provides JPopupTextMenu(owner) { textManager, items ->
+            JPopupMenu().apply {
+                textManager.cut?.also {
+                    add(
+                        swingItem(localization.cut, Color.RED, KeyEvent.VK_X, it)
+                    )
+                }
+                textManager.copy?.also {
+                    add(
+                        swingItem(localization.copy, Color.GREEN, KeyEvent.VK_C, it)
+                    )
+                }
+                textManager.paste?.also {
+                    add(
+                        swingItem(localization.paste, Color.BLUE, KeyEvent.VK_V, it)
+                    )
+                }
+                textManager.selectAll?.also {
+                    add(JPopupMenu.Separator())
+                    add(
+                       swingItem(localization.selectAll, Color.BLACK, KeyEvent.VK_A, it)
+                    )
+                }
+                for (item in items) {
```

Comment:
The `items` is received here from somewhere else.  As I understand, `items` is defined in `JPopupTextMenu`:
```kotlin
ContextMenuArea(items = { emptyList() }, state, content = content)
```
So it's hardcoded to be empty and  when using JPopupTextMenu directly, we can't change it. Maybe using `items` in this example could confuse a bit, since we can't change it? 

Maybe it's a question to the PR in androidx, but anyway:
Does `JPopupTextMenu` need List<ContextMenuItem>  in `private val createMenu: (TextManager, List<ContextMenuItem>) -> JPopupMenu` if the list is always empty?

Do I miss something?

Reply:
This tutorial is mostly about how to change appearance for all text context menus in the application, not only in some part of it. And when we change it on the global level, implementation of TextContextMenu should cover the case, when the user pass custom items down to all context menus:

```
         ContextMenuDataProvider(
            items = {
                listOf(
                    ContextMenuItem("User-defined Action") {/*do something here*/ },
                    ContextMenuItem("Another user-defined action") {/*do something else*/ }
                )
            }
        ) {
            TextField(
                value = text.value,
                onValueChange = { text.value = it },
                label = { Text(text = "Input") }
            )

            Spacer(Modifier.height(16.dp))

            SelectionContainer {
                Text("Hello World!")
            }
        }
```
Even if we pass emptyList in `JPopupTextMenu`, it will be merged with the other items, passed via ContextMenuDataProvider

Reply:
I see. Thanks